hack/run-hive-locally.sh

```diff
@@ -0,0 +1,349 @@
+    echo "The script runs a Hive component ($hive_component_operator or $hive_component_controllers) locally
+against the active cluster, simulating in-cluster environment. It records the in-cluster Hive
+configurations, scales down the selected component, then runs it locally with the same environment.
+On pressing ctrl + c, the script is terminated and relevant components are scaled back up.
+
+Warnings:
+- The active cluster may be in use, in which case the script could interfere with its functioning.
+
+Notes:
+- Artifacts (ConfigMaps containing in-cluster Hive configurations and log files) produced are stored in
+the --artifacts_folder which is removed once the script ends unless --preserve-artifacts is specified.
+- Selecting $hive_component_controllers runs all Hive controllers (including hive-clustersync) together
+as a single replica.
+- Endpoints of the Hive component running locally (e.g. metrics, pprof) are directly exposed on localhost.
+
+Usage:
+- ./hack/run-hive-locally.sh COMPONENT [--flag=value ...]
+
+Parameters:
+- COMPONENT: must be either $hive_component_operator or $hive_component_controllers
+
+Optional flags:
+- --hive-operator-ns=hive: namespace where the hive-operator is installed
+- --log-level=debug: log level of the component to run
+- --artifacts-folder=artifacts: folder that will be created to store the artifacts
+- --preserve-artifacts: if specified, preserve the artifacts folder after the script ends
+- --help"
+}
+
+function kill_background_jobs {
+    local pid
+    while read -r pid; do
+        hl_echo "Killing process $pid"
+        kill "$pid" || true
+    done < <(jobs -p)
+    wait
+}
+
+function get_envs {
+    local type_name="$1"
+    local ns="$2"
+    # Arrange output into the following format:
+    # name1 value1
+    # name2 value2
+    # ...
+    local jsonpath_query='{range .spec.template.spec.containers[0].env[*]}{@.name} {@.value}{"\n"}{end}'
+    local env
+    local value
+    local resolved_value
+
+    hl_echo "Getting environment variables from $type_name in ns/$ns"
+    while read -r env value; do
+        # Skip environment variables that are already resolved
+        if [ ${resolved_envs["$env"]+_} ]; then
+            hl_echo "Environment variable $env is already resolved, skipping"
+            continue
+        fi
+
+        # Skip irrelevant environment variables
+        if [[ $env = "CLI_CACHE_DIR" ]] || [[ $env = "HIVE_SKIP_LEADER_ELECTION" ]]; then
+            hl_echo "Environment variable $env is irrelevant, skipping"
+            continue
+        fi
+
+        # Resolve the environment variable's value
+        case "$value" in
+        /*)
+            resolved_value="./$artifacts_folder/${hc_spec_generation}$(tr / - <<< "$value").txt"
+            oc exec "$type_name" -n "$ns" -- /bin/bash -c "[ -f $value ] && cat $value || echo -n '!ENOENT!'" > "$resolved_value"
+            if [ '!ENOENT!' == "$(cat "$resolved_value")" ]; then
+                rm "$resolved_value"
```

Comment:
Removed `continue` after this line. This way in case the file does not exist, we still have ENV=FILE where FILE does not exist. This aligns with what we have in-cluster.

Reply:
Yah, cool.

May consider printing a warning about it.

Reply:
Noted. Will fix in future PR.